idseq_dag/steps/generate_phylo_tree.py

```diff
@@ -82,33 +90,60 @@ def run(self):
         #     shared exclusively by the descendants of that node.
         # Note: for integration with idseq-web, the node names need to be the pipeline_run_ids. So if we wanted to use outputs (2)/(3)/(4),
         # we would need to parse the appended information out from the newick node names and put it in a separate data structure.
-        command.execute(f"cd {self.output_dir_local}; mkdir ksnp3_outputs; kSNP3 -in inputs.txt -outdir ksnp3_outputs -k 13")
+        ksnp_cmd = (f"cd {self.output_dir_local}; mkdir ksnp3_outputs; "
+                    f"kSNP3 -in inputs.txt -outdir ksnp3_outputs -k 13")
+        if os.path.isfile(annotated_genome_input):
+            ksnp_cmd += f" -annotate {os.path.basename(annotated_genome_input)}"
+            # Note: produces SNP annotation file in a human-readable format that's very space inefficient (~100 MB).
+            # May want to do some postprocessing once we know better what exactly we need for the web app.
+        command.execute(ksnp_cmd)
         command.execute(f"mv {self.output_dir_local}/ksnp3_outputs/tree.parsimony.tre {output_files[0]}")
+        annotation_output = f"{self.output_dir_local}/ksnp3_outputs/SNPs_all_annotated"
+        if os.path.isfile(annotation_output):
+            try:
+                command.execute(f"mv  {self.output_dir_local}/ksnp3_outputs/SNPs_all_annotated {output_files[1]}")
+            except:
+                # HACK because web still makes the old DAG. Remove after new DAG is deployed.
+                command.execute(f"aws s3 cp {self.output_dir_local}/ksnp3_outputs/SNPs_all_annotated {os.path.join(self.output_dir_s3, 'SNP_annotations.txt')}")
```

Comment:
you can use the self.additional_files_to_upload.append('your file') for this purpose so it's non-blocking

Reply:
Oh great, thanks